cinema/permissions.py

```diff
@@ -0,0 +1,64 @@
+from rest_framework.exceptions import NotFound, MethodNotAllowed
+from rest_framework.permissions import BasePermission, SAFE_METHODS
+
+
+class IsAdminOrIfAuthenticatedReadOnly(BasePermission):
+    """The request is authenticated as an admin - read/write,
+    if as a user - read only request"""
+
+    def has_permission(self, request, view):
+        view_name = view.__class__.__name__
+        view_action = view.action
+
+        if view_name in [
+            "ActorViewSet",
+            "GenreViewSet",
+            "CinemaHallViewSet",
+        ] and view_action in ["retrieve", "destroy", "update"]:
+            raise NotFound("Object not found.")
```

Comment:
It's better to use `PermissionDenied` instead of `NotFound`

Reply:
`PermissionDenisd`  has code 403, but in tests we need 404